js/controllers/grid/articleGalleys/ArticleGalleyGridHandler.js

```diff
@@ -58,11 +58,18 @@
 		// FIXME: Inter-widget messaging is needed here.
 		var selector = 'a[id^="component-grid-articlegalleys-articlegalleygrid-row-' +
 				rowId + '-addFile-button-"]';
-		$.when($(selector)).then(function() {
-			$(function() {
+
+		function defferedClick(selector) {
+			if ($(selector).length) {
 				$(selector).click();
-			});
-		});
+			} else {
+				setTimeout(function() {
```

Comment:
I think this removes the example I cited earlier on the pkp-lib PR :) I liked the when/then formulation better; is there something here that breaks it?